src/plugins/charts/public/static/color_maps/color_maps.ts

```diff
@@ -54,3128 +62,87 @@ export interface ColorMap {
   [key: string]: RawColorSchema;
 }
 
+const COLOR_MAP_LENGTH = 512;
+
+function convertColorPaletteToColorMap(colorPalette: string[]): RawColorSchema['value'] {
+  const colorMap: RawColorSchema['value'] = [];
+
+  for (let i = 0; i < colorPalette.length; i++) {
+    const color = colorPalette[i];
+    const regex = /#([0-9a-zA-Z]{2})([0-9a-zA-Z]{2})([0-9a-zA-Z]{2})/;
+
+    // @ts-ignore
+    const [, rawRed, rawGreen, rawBlue] = regex.exec(color);
```

Comment:
Reviewers - the `ts-ignore` here is because Typescript doesn't like spreading the result of the regex for some reason. The error is something along the lines of `... doesn't implement [Iterator.Symbol]()`. I can change this to some other way of extracting these color parts if needed :)

Reply:
Oh I think I met this typescript error before...  I remember one solution is that you assign the result of `regex.exec(color)` to a var, then check null before destructuring. something like:

```
function convertColorPaletteToColorMap(colorPalette: string[]): RawColorSchema['value'] {
  const colorMap: RawColorSchema['value'] = [];

  for (let i = 0; i < colorPalette.length; i++) {
    const color = colorPalette[i];
    const regex = /#([0-9a-zA-Z]{2})([0-9a-zA-Z]{2})([0-9a-zA-Z]{2})/;

    const match = regex.exec(color);
    if (match) {
      const [, rawRed, rawGreen, rawBlue] = match;
      const [red, green, blue] = [
        parseInt(rawRed, 16) / 255,
        parseInt(rawGreen, 16) / 255,
        parseInt(rawBlue, 16) / 255,
      ];

      colorMap[i] = [i / (colorPalette.length - 1), [red, green, blue]];
    }
  }

  return colorMap;
}
```

Reply:
Thanks! Changed it to `const [, rawRed, rawGreen, rawBlue] = regex.exec(color)!;` because we know it will always be a color value, no need to explicitly null check :)